### PR TITLE
feat: Enforce keyword args for optional arguments.

### LINKED ISSUE #255: Use keywords for optional args in public methods.
For example, 

`def list_push_front(cache_name: str, list_name: str, value: str|bytes, *, ttl: CollectionTtl = CollectionTtl = CollectionTtl.from_cache_ttl(), truncate_back_to_size: Optional[int] = None)`

This forces the user to use keywords for the optional arguments making the order of arguments moot.

```
# fine
client.list_push_front(cname, lname, v)

# list_push_front() takes 3 positional arguments but 4 were given
client.list_push_front(cname, lname, v, ttl)

# fine
client.list_push_front(cname, lname, v, ttl=ttl)

# fine
client.list_push_front(cname, lname, v, truncate_back_to_size=ttl)

# fine
client.list_push_front(cname, lname, v, truncate_back_to_size=ttl, ttl=ttl)

# fine
client.list_push_front(cname, lname, v, ttl=ttl, truncate_back_to_size=ttl)
```

### PR DESCRIPTION
**Pull request by schwern**
This avoids having to worry about argument order.

I left set and set_if_not_exists alone becausae it would be a backwards incompat change.

Also
* set_add_elements() and set_remove_elements() was being passed a list, not a set.

Closes #255

### CODE REVIEW

**Comment by malandis** [comment]
> set_add_elements() and set_remove_elements() was being passed a list, not a set.

Why is this an issue?

**Comment by schwern** [comment]
@malandis 

> > set_add_elements() and set_remove_elements() was being passed a list, not a set.
> 
> Why is this an issue?

Technically it isn't, but it gave me pause.

**Comment by malandis** [comment]
> @malandis
> 
> > > set_add_elements() and set_remove_elements() was being passed a list, not a set.
> > 
> > 
> > Why is this an issue?
> 
> Technically it isn't, but it gave me pause.

If you want a really low overhead, trivial iterable, you can write: (element,). That is a 1-tuple which is iterable.

File: src/momento/simple_cache_client.py [hunk 1]
```diff
@@ -682,7 +690,7 @@ def set_add_element(
             CacheSetAddElementResponse
         """
 
-        resp = self.set_add_elements(cache_name, set_name, [element], ttl)
+        resp = self.set_add_elements(cache_name, set_name, {element}, ttl=ttl)
```
Comment (malandis) [inline src/momento/simple_cache_client.py:693]:
```suggestion
        resp = self.set_add_elements(cache_name, set_name, (element,), ttl=ttl)
```

File: src/momento/simple_cache_client.py [hunk 2]
```diff
@@ -741,7 +750,7 @@ def set_remove_element(self, cache_name: str, set_name: str, element: str | byte
         Returns:
             CacheSetRemoveElementResponse
         """
-        resp = self.set_remove_elements(cache_name, set_name, [element])
+        resp = self.set_remove_elements(cache_name, set_name, {element})
```
Comment (malandis) [inline src/momento/simple_cache_client.py:753]:
```suggestion
        resp = self.set_remove_elements(cache_name, set_name, (element,))
```

**Review by malandis — COMMENTED** [review]
If you want a really low overhead, trivial iterable, you can write: `(element,)`. That is a 1-tuple which is iterable.

**Review by malandis — APPROVED** [review]

**Pull request merged**